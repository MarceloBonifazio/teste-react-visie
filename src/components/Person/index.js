import React from 'react';
import PropTypes from 'prop-types';

import DeleteIcon from '@material-ui/icons/Delete';
import IconButton from '@material-ui/core/IconButton';
import Paper from '@material-ui/core/Paper';

import useStyles from './styles';

const Person = ({ color, name, removeItem, id }) => {
  const styles = useStyles();
  return (
    <Paper className={styles.root} style={{ color }}>
      <IconButton aria-label="Remove" onClick={() => removeItem(id)}>
        <DeleteIcon />
      </IconButton>
      {name}
    </Paper>
  );
};

Person.propTypes = {
  color: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string,
  removeItem: PropTypes.func.isRequired,
};

Person.defaultProps = {
  color: '',
  id: '',
  name: '',
};

export default Person;
