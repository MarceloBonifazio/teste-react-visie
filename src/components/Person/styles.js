import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  root: {
    paddingRight: theme.spacing(1),
    backgroundColor: '#ffe6e6',
    margin: '10px 0',
  },
}));

export default useStyles;
