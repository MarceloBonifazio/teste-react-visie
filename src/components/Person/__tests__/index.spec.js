import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import faker from 'faker';
import Person from '../index';

let removeItem = jest.fn();

describe('Person', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Should component render correctly...', () => {
    it('Without data', () => {
      const { baseElement } = render(<Person removeItem={removeItem} />);
      expect(baseElement).toMatchSnapshot();
    });

    it('With some data', () => {
      const { baseElement, getByText } = render(
        <Person
          name="Marcelo Bonifazio"
          color="black"
          id="5"
          removeItem={removeItem}
        />
      );
      expect(baseElement).toMatchSnapshot();
      expect(getByText(/Marcelo Bonifazio/)).toBeInTheDocument();
      expect(getByText(/Marcelo Bonifazio/)).toHaveStyle('color: black');
    });

    it('With another data', () => {
      const { baseElement, getByText } = render(
        <Person name="Teste" color="blue" id="5" removeItem={removeItem} />
      );
      expect(baseElement).toMatchSnapshot();
      expect(getByText(/Teste/)).toBeInTheDocument();
      expect(getByText(/Teste/)).toHaveStyle('color: blue');
    });

    it('Action should remove correctly element', () => {
      const id = faker.random.number(999999999).toString();

      removeItem = value => {
        expect(value).toStrictEqual(id);
      };

      const { baseElement, getByText } = render(
        <Person name="Teste" color="blue" id={id} removeItem={removeItem} />
      );
      expect(baseElement).toMatchSnapshot();
      fireEvent.click(getByText(/Teste/).parentNode.querySelector('button'));
    });
  });
});
