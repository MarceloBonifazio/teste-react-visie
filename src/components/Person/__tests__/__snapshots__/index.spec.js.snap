// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Should component render correctly... Action should remove correctly element 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root makeStyles-root-181 MuiPaper-elevation1 MuiPaper-rounded"
      style="color: blue;"
    >
      <button
        aria-label="Remove"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      Teste
    </div>
  </div>
</body>
`;

exports[`Person Should component render correctly... With another data 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root makeStyles-root-121 MuiPaper-elevation1 MuiPaper-rounded"
      style="color: blue;"
    >
      <button
        aria-label="Remove"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      Teste
    </div>
  </div>
</body>
`;

exports[`Person Should component render correctly... With some data 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root makeStyles-root-61 MuiPaper-elevation1 MuiPaper-rounded"
      style="color: black;"
    >
      <button
        aria-label="Remove"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      Marcelo Bonifazio
    </div>
  </div>
</body>
`;

exports[`Person Should component render correctly... Without data 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        aria-label="Remove"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      
    </div>
  </div>
</body>
`;
